src/Data/Maybe.ard

```diff
@@ -0,0 +1,52 @@
+\import Logic
+\import Paths
+\import Set
+
+-- | Maybe A
+-- Maybe represents an optional value. It might also be used to represent failure.
+\data Maybe (A : \Type) | Nothing | Just A
+  \where {
+    \func bind {A B : \Type} (a : Maybe A)(f : A -> Maybe B) : Maybe B \elim a
+      | Nothing => Nothing
+      | Just a => f a
+
+    \lemma JustA/=Nothing  {A : \Type} (a : A) (p :  Just a = Nothing) : Empty =>
+      IsJust.absurd (transport IsJust p ItIsJust)
```

Comment:
`\elim p | ()`?

Reply:
Didn't get this.

Reply:
Arend has absurd patterns. You don't need to write all the absurd eliminators add as functions.

Reply:
Cool. Is there an example of absurd pattern somewhere?

Here it doesn't look like a complete absurd. `Path` has `path` constructor.

I found a similar example here: https://github.com/JetBrains/arend-lib/blob/master/src/Data/Bool.ard#L23 . Is your advice applicable there as well?

Reply:
Ah I missed that. Then it works.